cdap-proto/src/main/java/co/cask/cdap/proto/AdapterDetail.java

```diff
@@ -16,24 +16,28 @@
 
 package co.cask.cdap.proto;
 
-import com.google.common.base.Objects;
+import co.cask.cdap.api.schedule.ScheduleSpecification;
 import com.google.gson.JsonElement;
 
 /**
- * Adapter details that come back from get calls.
- * TODO: finalize what should be present here
+ * Represents an adapter returned for /adapters/{adapter-id}.
  */
-public final class AdapterDetail {
+public class AdapterDetail {
   private final String name;
   private final String description;
   private final String template;
+  private final Id.Program program;
```

Comment:
What does it look like for the serialized object? It seems like it'd be pretty nested and not good to be used as response of the REST endpoint. It looks something like this:

```
{
  "application": {
    "namespace": {
      "id": "ns"
    },
    "applicationId": "appId"
  },
  "type": "Worker",
  "id": "pid"
}
```